File: jna-jni-example/src/main/java/com/github/jjYBdx4IL/maven/examples/jnajni/JNAAccess.java
package com.github.jjYBdx4IL.maven.examples.jnajni;

import com.sun.jna.Native;
import com.sun.jna.NativeLibrary;

/**
 * Access native function through JNA.
 *
 * @author jjYBdx4IL
 */
public class JNAAccess {

    static {
        NativeLibrary lib = NativeLibrary.getInstance("jnaaccess");
        Native.register(lib);
    }

    public static native int sum(int[] intArr, int arrayLength);
    public static native int inc(int arg);
}

File: jna-jni-example/src/main/c/jnaaccess.c
// test functions for native access via JNA

int sum (int *arr, int arrLength)
{
    int sum = 0;
    int i;

    for (i=0; i < arrLength; i++)
        sum += arr[i];

    return sum;
}

int inc(int arg)
{
    return arg + 1;
}